### DISCUSSION TITLE
Trigger/ non-trigger  or edge function to compress and resize images on the fly and with cron job

### DISCUSSION THREAD

**Discussion by organicnz** [supabase/supabase#12316] [Questions] [answered]
Hi folks,

Could we, by any chance, implement a trigger/ non-trigger or edge function to compress and resize images on the fly and with corn job? It would be great to hear if someone has already implemented such a feature :))

Thank you :)

**Comment by Joeao** [comment, marked as answer]
I made an edge function to resize, optimise and store an image. I use it instead of uploading directly to the bucket. It doesn't do exactly what you want, but may provide some use to you or others wanting to achieve something similar.

https://gist.github.com/Joeao/240ba519b8bc9cf5090f81e173716656

**Reply by organicnz** [reply]
Thanks so much for sharing, mate. I'll try to implement your edge function :)

**Comment by laktek** [comment]
@Joeao This looks pretty handy! Do you want to contribute it to Supabase Edge Functions docs as an example (we will credit you) Docs live here https://github.com/supabase/supabase/tree/master/apps/docs/pages/guides/functions/examples

**Reply by Joeao** [reply]
Done https://github.com/supabase/supabase/pull/18039
